adapters/repos/db/vector/hnsw/search.go

```diff
@@ -210,6 +216,13 @@ func (h *hnsw) searchLayerByVectorWithDistancer(queryVector []float32,
 	connectionsReusable := make([]uint64, h.maximumConnectionsLayerZero)
 
 	for candidates.Len() > 0 {
+		time.Sleep(1 * time.Second)
```

Comment:
This adds an effective one second sleep in what looks like the core loop of the search, are we ok with that ?

Cancellation looks OK to me :)

Reply:
Thanks! That was of course not intentional. The last commit added this (it was local change for testing). Removed again!